src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -938,12 +939,11 @@ private bool UpdateDatabaseStateAfterSync()
                 [Conditional("DEBUG")]
                 private void AssertGatherInformationToStartSyncBeforeUpdate()
                 {
-                    if (_lastSyncedJournal == -1
-                        && _currentTotalWrittenBytes == -1
-                        && _lastSyncedTransactionId == -1)
+                    if (_lastFlushed == null && _currentTotalWrittenBytes == -1)
+                    {
                         throw new InvalidOperationException(
-                            $"Try to {nameof(UpdateDatabaseStateAfterSync)} " +
-                            $"without to call {nameof(GatherInformationToStartSync)} before");
+                            $"Try to {nameof(UpdateDatabaseStateAfterSync)} without to call {nameof(GatherInformationToStartSync)} before");
```

Comment:
Without "calling", not "to call"

Reply:
Fixed in https://github.com/ravendb/ravendb/pull/8424